File: Model/Actor.fs
namespace BattleCity.Model

open Avalonia
open BattleCity.Infrastructure

[<AbstractClass>]
type GameObject internal (location: Point) =
    inherit PropertyChangedBase()
    let mutable _location = location

    member my.Location
        with get() = _location
        and internal set (v) = 
            my.SetProperty(& _location, v) |> ignore

    abstract Layer: int with get
    default _.Layer = 0

File: Model/Actor.cs
using BattleCity.Infrastructure;
using Avalonia;
using CommunityToolkit.Mvvm.ComponentModel;

namespace BattleCity.Model;

public abstract class GameObject : ObservableObject
{
    private Point _location;

    protected GameObject(Point location)
    {
        Location = location;
    }

    public Point Location
    {
        get => _location;
        protected set => SetProperty(ref _location, value);
    }

    public virtual int Layer => 0;
}